lib/modules/manager/vendir/extract.ts

```diff
@@ -0,0 +1,78 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { parseSingleYaml } from '../../../util/yaml';
+import { DockerDatasource } from '../../datasource/docker';
+import { HelmDatasource } from '../../datasource/helm';
+import type { PackageDependency, PackageFileContent } from '../types';
+import type { HelmChart, Vendir } from './types';
+import { isHelmChart, isOCIRegistry } from './utils';
+
+// TODO: Add support for other vendir types (like git tags, github releases, etc.)
+// Recommend looking at the kustomize manager for more information on support.
+
+export function extractHelmChart(
+  helmChart: HelmChart,
+): PackageDependency | null {
+  let registryUrl = helmChart.repository.url;
+  let dataSource = HelmDatasource.id;
+  if (isOCIRegistry(helmChart.repository)) {
+    registryUrl = helmChart.repository.url.replace('oci://', 'https://');
```

Comment:
support registry alias, see kustomize manager for implementation

Reply:
Added support for registryAliases for OCI and https repos with accompanying test